.github/workflows/matter-validation.yml

```diff
@@ -0,0 +1,68 @@
+name: Matter Validation
+
+on:
+  pull_request:
+    branches:
+      - main
+      - 'v*-branch'
+
+    paths:
+      - 'samples/matter/**'
+      - 'applications/matter_bridge/**'
+      - 'applications/matter_weather_station/**'
+
+permissions:
+  contents: read
+
+jobs:
+  matter-validation:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v5
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+          path: ncs/nrf
+
+      - name: Rebase
+        if: github.event_name == 'pull_request'
+        continue-on-error: true
+        env:
+          BASE_REF: ${{ github.base_ref }}
+          PR_HEAD: ${{ github.event.pull_request.head.sha }}
+        working-directory: ncs/nrf
+        run: |
+          git config --global user.email "actions@zephyrproject.org"
+          git config --global user.name "Github Actions"
+          rm -fr ".git/rebase-apply"
+          rm -fr ".git/rebase-merge"
+          git rebase origin/${BASE_REF}
+          git clean -f -d
+          git log --graph --oneline HEAD...${PR_HEAD}
+
+      - name: Set up Python
+        uses: actions/setup-python@v6
+        with:
+          python-version: 3.12
+          cache: pip
+          cache-dependency-path: ncs/nrf/scripts/requirements.txt
+
+      - name: Install packages
+        working-directory: ncs/nrf
+        run: |
+          sudo apt update
+          sudo apt-get install -y wget python3-pip git
+          pip install -r scripts/requirements.txt
+
+      - name: Checkout Matter repository
+        working-directory: ncs
+        run: |
+          git clone https://github.com/nrfconnect/sdk-connectedhomeip modules/lib/matter
```

Comment:
Why don't you use west update? 
Could you determine proper branch for legacy releases?
nrf's `main` -> matter's `master`
nrf's `vX.Y-branch` ->matter's `vX.Y-branch`

Reply:
I gave up with west update because it fetches all redundant repository whereas I need to have just SHA of current sdk-connectedhomeip revision. 

Good point. I will add branch slection.